memreps/memreps.py

```diff
@@ -71,12 +91,27 @@ def find_distiguishing(concept1, concept2, atoms):
             assert (y in concept2) == polarity
             return x, y
 
+def find_maximally_distinguishing(concept_gen, max_concepts=10, atoms_per_concept=1):
+    concepts = fn.take(max_concepts, concept_gen())
+    pairs = combinations(concepts, 2)
+    if len(concepts) == 2: # pairs doesn't seem to return a single pair when len is 2?
+        pairs = [tuple(concepts)]
+        atoms_per_concept = max(2, atoms_per_concept)
+    diffs = (c1 ^ c2 for c1, c2 in pairs)
+    atoms = (set(fn.take(atoms_per_concept, c)) for c in diffs)
+    atoms = set.union(*atoms)
+    atoms = sorted(atoms, key=lambda x: min(estimate_reductions(('∈', x), concepts).values()))
+    prefqueries = list(combinations(atoms, 2))
+    prefqueries = sorted(prefqueries, key=lambda x: min(estimate_reductions(('≺', x), concepts).values()))
```

Comment:
Is this being optimistic?

Reply:
That's not a bad thing. Just want to be clear.

Also it might be nice to make separate key helper functions that specialize to the membership and preference cases. 
This let's us also make sure to ignore the incomparable case right (if we go to worst case, e.g. `max`).

this may be helpful:
```python
mapping = fn.omit(mapping, ['||'])  # ignores ||.
```

Reply:
hmmmmmmm yeah it is being optimistic but i'd agree that going to worst-case makes more sense. i can make this change